File: docs/src/testers/functions/invocations/stream_sdk.py
# @sniptest filename=stream_sdk.py
# @sniptest show=8-12
import os

from notte_sdk import NotteClient

client = NotteClient()
function = client.Function(os.environ["NOTTE_FUNCTION_ID"])

# Stream logs while running
result = function.run(
    url="https://example.com",
    stream=True,  # Logs printed to console
)

File: docs/src/testers/functions/invocations/stream_sdk.ts
// @sniptest filename=stream_sdk.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
const fn = client.NotteFunction({
  function_id: process.env.NOTTE_FUNCTION_ID!,
});

// Stream logs while waiting for the final result.
const result = await fn.run({ url: 'https://example.com' }, { stream: true });
console.log(JSON.stringify({ status: result.status, result: result.result }));
